src/apps/sketch/communication/HG.js

```diff
@@ -152,8 +152,13 @@ export class Guest {
         console.log({ channel });
         this.channels[channel.label] = channel;
         // channel.onmessage = message => console.log({ channel: message.currentTarget.label, message: message.data });
-        if (Object.keys(this.channels).length >= 4) {
-            this.onConnection()
```

Comment:
I added this so that I can confirm that all channels are created before starting the session, so that there wont be any channel undefined errors later

Reply:
don't worry I am going to use this line at some other place, we won't be able to start session if all the data channels aren't available